src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -2285,6 +2285,39 @@ Address MacroAssembler::form_address(Register Rd, Register base, int64_t byte_of
   return Address(Rd);
 }
 
+// On Neoverse, MSUB uses the same ALU with other instructions (e.g. SDIV).
+// The combination of MUL/SUB can utilize multiple ALUs,
+// and can be somewhat faster than MSUB.
+void MacroAssembler::msub(Register Rd, Register Rn, Register Rm, Register Ra,
+                          Register scratch)
+{
+  assert(Rn != scratch && Rm != scratch && Ra != scratch, "reg cannot be scratch");
+
+  if (VM_Version::supports_a53mac() && Ra != zr)
+    nop();
+  if (VM_Version::is_neoverse()) {
```

Comment:
Maybe a more descriptive name?
```suggestion
  if (VM_Version::split_msub()) {
```

Is there a good reason why we're restricting this to msub and not madd?